### DISCUSSION TITLE
Contributing XCBBuildServiceProxy

### DISCUSSION THREAD

**Discussion by pennig** [MobileNativeFoundation/discussions#110] [Proposals]
Target has open-sourced an internally-built tool called [XCBBuildServiceProxy](https://github.com/target/XCBBuildServiceProxy) which acts as a proxy between Xcode and XCBuild to facilitate Bazel integration with Xcode. I'd like to start the discussion on whether it would be valuable and desirable to be contributed to the MNF. I can start the discussion by responding to the criteria proposed in [this discussion](https://github.com/MobileNativeFoundation/discussions/discussions/46). If there's a more complete way to propose a contribution, please let me know!

1. The project should be open source and under a permissible license
✅ XCBBuildServiceProxy carries an Apache 2.0 license.

2. There should be a community interest in the project
✅ Bazel is proving to be a popular tool among large teams developing complex apps. Making it easier to integrate Bazel into one's organization has a strong community interest.

3. There should be a suitable scope for the community
🤷 This project is intended only for large teams who are struggling with long compile and testing times due to project size and complexity, so the scope is significantly reduced, but I think the utility of it helps to balance it out.

4. The project should be non-trivial
✅ Inarguably non-trivial, containing thousands of lines of code.

5. The adoption should serve a purpose
✅ The original contributor(s) no longer work at Target, and while the project is being maintained by my team, it really needs contributions from the broader community to keep up with Xcode updates, and to address existing limitations.

6. The project should be "production ready"
✅ The project is continues to be used by Target, so is definitely deemed production ready.

7. The existing maintainers should ideally want to continue in that role
🤷 The existing maintainers are admittedly not fully equipped to continue maintaining it, at least to the degree the project deserves. But it's exactly for that reason I'm proposing this contribution—the _original_ maintainer(s) have expressed interest in continuing working on this project. 

We should keep in mind that Target may not be willing to contribute this project (regrettable though it may be), but I still wanted to determine whether there's interest here first, before engaging with our open source folks and lawyers.

**Comment by brentleyjones** [comment]
I wanted to pop in fast to say I’m the original maintainer of the project, and to speak to bullet 7, I would be willing to resume maintainership if it was moved.

**Comment by keith** [comment]
I'm supportive of this project moving to the MNF, I think it's something folks are definitely interested in using once they're into bazel and want to get as close to the Xcode experience as possible. I think it makes sense to try and align the community on this 1 implementation. I'm definitely interested to hear other's thoughts here as well.

**Reply by ob** [reply]
I'm also supportive. This is definitively one of the pain points of building locally with Bazel.

**Reply by rudro** [reply]
Same. Also supportive 👍. We don't use this project (currently looking at VS Code as an alternative) but it does seem like it would be broadly useful to most projects using bazel. I'm definitely interested in trying this out with our codebase.

**Reply by dalemyers** [reply]
Yeah, this covers a huge pain point that has been seen over many different technologies, not just Bazel. I'd love to see this become a de facto tool.

**Comment by erikkerber** [comment]
(As part of the team that first used this) Supportive and willing to contribute myself 👍🏼

**Comment by BalestraPatrick** [comment]
This project lookks very interesting and I would be happy to see it develop further. It's a 👍 from my side too.

**Comment by shepting** [comment]
This would be great!

**Comment by brentleyjones** [comment]
Today Target has donated XCBBuildServiceProxy to the MNF: https://github.com/MobileNativeFoundation/XCBBuildServiceProxyKit!

**Reply by keith** [reply]
Thanks everyone!
